src/rasterstats/main.py

```diff
@@ -195,6 +197,14 @@ def gen_zonal_stats(
                         feature_stats = remap_categories(category_map, feature_stats)
                 else:
                     feature_stats = {}
+                
+            
+                if 'properties' in feat:
+                    if preserve_properties:
+                        feature_stats['properties'] = feat['properties']
+                    if preserve_ids:
+                        if 'id' in feat['properties']:
+                            feature_stats['id'] = feat['properties']['id']
```

Comment:
Three suggestions for this section:

1. We can simplify the branching logic here

```
if preserve_properties and 'properties' in feat:
    ...
```

2. If the id is already part of properties, this argument is superfluous. The true id, according to the GeoJSON spec, is at the top level of the feature.

```
if preserve_ids and 'id' in feat:
    feature_stats['id'] = feat['id']
```

3. Because this code deals with massaging the output data, not directly with zonal stats itself, I'd like to move it to the bottom  of the function.  This should share code with the application of `geojson_out` (geojson_out is basically equivalent to `preserve_properties` + `preserve_ids` + `preserve_geometry`)

Reply:
github decided not to give any notifications for this so i've only just noticed the comments. I'll look into making the changes as discussed.

thanks